ethdb/remotedb/remotedb.go

```diff
@@ -159,11 +162,23 @@ func dialRPC(endpoint string) (*rpc.Client, error) {
 		// these prefixes.
 		endpoint = endpoint[4:]
 	}
-	return rpc.Dial(endpoint)
+	var opts []rpc.ClientOption
+	if len(headers) > 0 {
+		var customHeaders = make(http.Header)
+		for _, h := range headers {
+			kv := strings.Split(h, ":")
+			if len(kv) != 2 {
+				return nil, fmt.Errorf("invalid http header directive: %q", h)
+			}
+			customHeaders.Add(kv[0], kv[1])
+		}
+		opts = append(opts, rpc.WithHeaders(customHeaders))
+	}
+	return rpc.DialOptions(context.Background(), endpoint, opts...)
 }
 
-func New(endpoint string) (ethdb.Database, error) {
-	client, err := dialRPC(endpoint)
+func New(endpoint string, headers []string) (ethdb.Database, error) {
+	client, err := dialRPC(endpoint, headers)
```

Comment:
Might be better to change this constructor to take `*rpc.Client`. Then you don't need to duplicate the dial options in this package.